src/index.ts

```diff
@@ -104,21 +111,50 @@ const showPrivacyManager = () => {
 	void initialised.then(frameworkCMP?.showPrivacyManager);
 };
 
-export const cmp: CMP = (window.guCmpHotFix.cmp ||= {
-	init,
-	willShowPrivacyMessage,
-	willShowPrivacyMessageSync,
-	hasInitialised,
-	showPrivacyManager,
-	version: __PACKAGE_VERSION__,
-
-	// special helper methods for disabling CMP
-	__isDisabled: isDisabled,
-	__enable: enable,
-	__disable: disable,
-});
+export const cmp: CMP = (() => {
+	if (isServerSide) {
+		return {
+			init: (_: { pubData?: PubData; country?: Country }) => void 0,
+			showPrivacyManager: () => {
+				void 0;
+			},
+			willShowPrivacyMessage: () => new Promise(() => false),
+			willShowPrivacyMessageSync: () => false,
+			hasInitialised: () => true,
+			version: __PACKAGE_VERSION__,
+
+			// special helper methods for disabling CMP
+			__isDisabled: isDisabled,
+			__enable: enable,
+			__disable: disable,
+		} as CMP;
+	}
 
-export const onConsentChange = (window.guCmpHotFix.onConsentChange ||=
-	actualOnConsentChange);
-export const getConsentFor = (window.guCmpHotFix.getConsentFor ||=
-	actualGetConsentFor);
+	return (window.guCmpHotFix.cmp ||= {
+		init,
+		willShowPrivacyMessage,
+		willShowPrivacyMessageSync,
+		hasInitialised,
+		showPrivacyManager,
+		version: __PACKAGE_VERSION__,
+
+		// special helper methods for disabling CMP
+		__isDisabled: isDisabled,
+		__enable: enable,
+		__disable: disable,
+	});
+})();
+
+export const onConsentChange = (() => {
+	if (isServerSide) {
+		return () => void 0;
+	}
+	return (window.guCmpHotFix.onConsentChange ||= actualOnConsentChange);
+})();
+
+export const getConsentFor = (() => {
+	if (isServerSide) {
+		actualGetConsentFor;
```

Comment:
should this be another one of these?

```suggestion
		return () => void 0;
```

Reply:
As far as I can tell the implementation of `getConsentFor` is stateless and already safe if it were without `window`.